src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -0,0 +1,389 @@
+        private readonly HttpClient _client;
+        private readonly HttpClientHandler _handler;
+        private readonly MemoryStream _messageStream;
+        private readonly Dictionary<string, string> _contentHeaderValueCache;
+
+        /// <summary>
+        /// Constructor for HttpClientRequestMessage.
+        /// </summary>
+        public HttpClientRequestMessage(DataServiceClientRequestMessageArgs args) 
+            : base(args.ActualMethod)
+        {
+            _requestMessage = new HttpRequestMessage();
+            _messageStream = new MemoryStream();
+            _handler = new HttpClientHandler();
+            _client = new HttpClient(_handler, disposeHandler: true);
+            _contentHeaderValueCache = new Dictionary<string, string>();
+
+            _effectiveHttpMethod = args.Method;
+            _requestUrl = args.RequestUri;
+
+            _requestMessage = new HttpRequestMessage(new HttpMethod(args.Method), _requestUrl);
+
+            // Now set the headers.
+            foreach (var keyValue in args.Headers)
+            {
+                this.SetHeader(keyValue.Key, keyValue.Value);
+            }
+        }
+
+        /// <summary>
+        /// Returns the collection of request headers.
+        /// </summary>
+        public override IEnumerable<KeyValuePair<string, string>> Headers
+        {
+            get
+            {
+                if (_requestMessage.Content != null)
+                {
+                    return HttpHeadersToStringDictionary(_requestMessage.Headers).Concat(HttpHeadersToStringDictionary(_requestMessage.Content.Headers));
+                }
+
+                return HttpHeadersToStringDictionary(_requestMessage.Headers).Concat(_contentHeaderValueCache);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the request url.
+        /// </summary>
+        public override Uri Url
+        {
+            get
+            {
+                return _requestMessage.RequestUri;
+            }
+            set
+            {
+                _requestMessage.RequestUri = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the method for this request.
+        /// </summary>
+        public override string Method
+        {
+            get
+            {
+                return _requestMessage.Method.ToString();
+            }
+            set
+            {
+                _requestMessage.Method = new HttpMethod(value);
+            }
+        }
+
+        /// <summary>
+        ///  Gets or set the credentials for this request.
+        /// </summary>
+        public override ICredentials Credentials
+        {
+            get
+            {
+                return _handler.Credentials;
+            }
+            set
+            {
+                _handler.Credentials = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the timeout (in seconds) for this request.

```

Comment:
Is this documentation for the If or for the Timeout? Perhaps the If also needs some documentation.

Reply:
@wachugamaina  it is for the timeout